compiler/semantic/op.go

```diff
@@ -418,13 +418,14 @@ func semOp(ctx context.Context, scope *Scope, o ast.Op, ds *data.Source, head *l
 			NullsFirst: o.NullsFirst,
 		}, nil
 	case *ast.Head:
-		limit := o.Count
-		if limit == 0 {
-			limit = 1
+		expr, _ := semExpr(scope, o.Count)
+		val, _ := kernel.EvalAtCompileTime(scope.zctx, expr)
```

Comment:
We should handle these errors.
```suggestion
		expr, err := semExpr(scope, o.Count)
		if err != nil {
			return nil, fmt.Errorf("head: %w", err)
		}		
		val, err := kernel.EvalAtCompileTime(scope.zctx, expr)
		if err != nil {
			return nil, fmt.Errorf("head: %w", err)
		}		
```